cmd/oras/root/attach.go

```diff
@@ -91,14 +95,15 @@ Example - Attach file to the manifest tagged 'v1' in an OCI layout folder 'layou
 
 	cmd.Flags().StringVarP(&opts.artifactType, "artifact-type", "", "", "artifact type")
 	cmd.Flags().IntVarP(&opts.concurrency, "concurrency", "", 5, "concurrency level")
+	cmd.Flags().BoolVarP(&opts.strict, "strict", "", false, "strictly handles any ignorable errors as fatal")
```

Comment:
@FeynmanZhou @yizha1 Any comments on the flag name?

Reply:
What does ignorable error mean? Is it a warning or error?

Reply:
> What does ignorable error mean?

Currently it means attach is successful but something went wrong. It should be only a warning if `--strict` is not set.

Reply:
Is this flag passed into the logger or used somewhere? 
Also `--strict` seems more like a type checking kind of usage. 
Make uses `--ignore-errors` which seems more inline with. The issue I see is that `--ignore-errors` needs to be true by default in this case for refererrer delete failure to be ignored, but othere errors might also exists which aren't ignorable. 

That said why do we even need this flag and can we introduce this when there is an ask to block on this error?

Reply:
> Is this flag passed into the logger or used somewhere?

It's not passed to the logger, it will be used to terminate code run if an exceptional error returned from cleaning the old referrer index.

If the index cleanup fails:
- If the return type is deletion not supported, then attaching succeeds
- Otherwise
   - if the `--strict` is set, then attaching fails
   - otherwise attaching succeeds
 
> can we introduce this when there is an ask to block on this error?

@shizhMSFT suggested adding this flag in an offline discussion and @yizha1 @FeynmanZhou agreed on adding it.

Reply:
@sajayantony @shizhMSFT @yizha1 @TerryHowe  The scenario of using the `--strict` flag in `oras attach` is when users want to disallow attaching referrers if the Deletion API is disabled by the registry or garbage collection is not supported by the registry. Without this flag, the registry might be filled with lots of outdated referrer index and eventually impact the registry performance. Does it make sense?

The flag name `--strict` might be a little vague. I am thinking if there is a better name to make it much more straightforward.

Reply:
Also `oras cp` need this flag: if referrer API is not supported and the index need to be updated during the copy, consider copying A in the below diagram:

```mermaid
graph TD;
  A--platform b-->B
  A--platform c-->C
  A--platform d-->D
  B--subject-->E
  C--subject-->E
  D--subject-->E
```

If the concurrency is set to 1, the referrer index of E will be updated twice.

Reply:
I'm still of the opinion that delete failure should be a warning. The concurrency issue seems like something that SDK should fix and not depend on put/delete semantics for consistency at least from the client's perspective. 

I still think we should avoid this flag all together since Manifest delete itself is quite a low level semantic and not something that the user should care about. It's up to the registry operators to GC them since they aren't tagged anymore and if delete is disabled on them. 

If we show a warning  `Delete failed for referrers index: sha256:... . `  that should be good enough for GHCR users or DH users to follow up with clean up if they can or let it dangle and be purged by the registry. 

> Without this flag, the registry might be filled with lots of outdated referrer index and eventually impact the registry performance. Does it make sense?
This is true even with the flag is my understanding and hence with or without the flag the outcome seems the same. The impact is that whenflag is passed we skip the delete manifest API otherwise its a warning. If we truely need a flag we should wait is my take.

Reply:
> If we show a warning something like WARNING: Delete failed for referrers index: sha256:... .  that should be good enough for GHCR users or DH users to follow up with clean up if they can or let it dangle and be purged by the registry.

@sajayantony I agree that we need to show the digest of referrers index. Will change accordingly.

Reply:
> I'm still of the opinion that delete failure should be a warning. The concurrency issue seems like something that SDK should fix and not depend on put/delete semantics for consistency at least from the client's perspective.
> 
> I still think we should avoid this flag all together since Manifest delete itself is quite a low level semantic and not something that the user should care about. It's up to the registry operators to GC them since they aren't tagged anymore and if delete is disabled on them.
> 
> If we show a warning something like `WARNING: Delete failed for referrers index: sha256:... . ` that should be good enough for GHCR users or DH users to follow up with clean up if they can or let it dangle and be purged by the registry.
> 
> > Without this flag, the registry might be filled with lots of outdated referrer index and eventually impact the registry performance. Does it make sense?
> 
> This is true even with the flag is my understanding and hence with or without the flag the outcome seems the same. The impact is that when the flag is passed we skip the delete manifest API otherwise it's a warning. If we truly need a flag we should wait is my take. Introducing this and removing this is not a good progression.

+1

Reply:
Agree with @sajayantony . We can add this flag until there is a real user who requests this advanced feature in the future. From the ORAS client side, reminding users in a warning log output about GC is enough at this moment.